server/gfmd/auth_uid.c

```diff
@@ -313,6 +344,46 @@ auth_uid_to_global_username_sharedsecret(void *closure,
 	return (GFARM_ERR_NO_ERROR);
 }
 
+#ifdef HAVE_CYRUS_SASL
+
+static gfarm_error_t
+auth_uid_to_global_username_sasl(void *closure,
+	const char *auth_user_id, enum gfarm_auth_id_role *auth_user_id_rolep,
+	char **global_usernamep)
+{
+	enum gfarm_auth_id_role auth_user_id_role = *auth_user_id_rolep;
+	char *global_username;
+	struct user *u;
+	const char diag[] = "auth_uid_to_global_username_sasl";
+
+	if (auth_user_id_role != GFARM_AUTH_ID_ROLE_USER)
+		return (GFARM_ERR_AUTHENTICATION);
```

Comment:
NOTE: this check (i.e. auth_user_id_role != GFARM_AUTH_ID_ROLE_USER) is good, because SASL is only  used for user authentication, and never used for gfsd authentication (GFARM_AUTH_ID_ROLE_SPOOL_HOST) and gfmd authentication (GFARM_AUTH_ID_ROLE_METADATA_HOST).